File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapNodes(ListNode head, int k) {
        ListNode leftNode = head;
        ListNode rightNode = head;
        for(int i=1; i<k; i++) leftNode = leftNode.next;
        ListNode curr = leftNode;
        while(curr.next != null){
            curr = curr.next;
            rightNode = rightNode.next;
        }
        int temp = leftNode.val;
        leftNode.val = rightNode.val;
        rightNode.val = temp;
        return head;
    }
}

File: 1721-swapping-nodes-in-a-linked-list/1721-swapping-nodes-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        leftNode, rightNode = head, head
        for i in range(1, k): leftNode = leftNode.next
        curr = leftNode
        while curr.next != None:
            curr = curr.next
            rightNode = rightNode.next
        temp = leftNode.val
        leftNode.val = rightNode.val
        rightNode.val = temp
        return head